### PR TITLE
Tutorial to create your own worker

### PR DESCRIPTION
**Pull request by landru29**

### CODE REVIEW

File: doc/tutorials/workers.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+## Worker cycle of life
+
+* The worker is launched in the docker image (or openstack, ...). So, first, the docker must be able to download it.
+* Then your stage script is executed.
+
+As a consequence, the worker model (Docker image) must:
+
+* have a sane installation of `curl`
+* specify a working dir on which the current user is allowed to write
```
Comment (sguiheux) [inline doc/tutorials/workers.md]: working directory

File: doc/tutorials/workers.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+* `grunt-cli`
+* `gulp-cli`
+* `fontforge`
+* `ttfautohint`
+
+We will use the official **nodejs** image from Docker. In this image, there is already a user named **node**. For the example, we will compile `ttfautohint`.
+
+### Prerequisite
+
+To build a model, you need:
```
Comment (sguiheux) [inline doc/tutorials/workers.md]: a docker model

File: doc/tutorials/workers.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+* a sane installation of Docker [https://docs.docker.com/engine/installation/](https://docs.docker.com/engine/installation/)
+
+### Dockerfile
+
+Copy this content in a file named `Dockerfile`
+
+```
+# User official nodejs docker image
+FROM node:6.10.1
+MAINTAINER Cyrille Meichel "cyrille.meichel@corp.ovh.com"
```
Comment (yesnault) [inline doc/tutorials/workers.md]: keyword MAINTENER deprecated

File: doc/tutorials/workers.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+#Answer 'yes' to each question
+ENV DEBIAN_FRONTEND noninteractive
+
+# Upgrade the debian packages
+RUN (apt-get update && apt-get upgrade -y -q && apt-get dist-upgrade -y -q && apt-get -y -q autoclean && apt-get -y -q autoremove)
```
Comment (yesnault) [inline doc/tutorials/workers.md]: dist-upgrade? -> not usefull I think.

File: doc/tutorials/workers.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+# Remember, a curl command will be, first, executed to download the worker
+WORKDIR /home/node
+```
+
+### Build it and push it
+
+from you shell, type the following command to build the Docker image:
+
+```
+docker build --no-cache --pull -t registry.myInfra.net/my/beautiful/worker:latest .
```
Comment (yesnault) [inline doc/tutorials/workers.md]: remove uppercase

File: doc/tutorials/workers.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+Now push it
+
+```
+docker push registry.myInfra.net/my/beautiful/worker:latest
+```
+
+### Register your model in CDS
+
+* In the UI, clic on the wheel on the hand right top corner and select *workers" (or go the the route *#/worker*)
```
Comment (yesnault) [inline doc/tutorials/workers.md]: clicKK

File: doc/tutorials/workers.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+* At the bottom of the page, fill the form
+    * Name of your worker *My Beautiful*
+    * type *docker*
+    * image *registry.myInfra.net/my/beautiful/worker:latest*
+* Clic on *Add* button and that's it
+
+Now you can specify this model in your pipelines
```
Comment (yesnault) [inline doc/tutorials/workers.md]: Now you can specify this model in prerequisite on your job. Create a new prerequisite of type "model", then choose your worker model in list

File: doc/tutorials/workers.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+To build a Docker model, you need:
+
+* your favorite text editor
+* a sane installation of Docker [https://docs.docker.com/engine/installation/](https://docs.docker.com/engine/installation/)
+
+### Dockerfile
+
+Copy this content in a file named `Dockerfile`
+
+```
```
Comment (yesnault) [inline doc/tutorials/workers.md]: Dockerfile content, with ```dockerfile -> this show color
```dockerfile
FROM example
```

File: doc/tutorials/workers.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
+# Workers
+
+**CDS** uses workers to execute pipelines. You will need, for sure, to create your own worker models.
```
Comment (yesnault) [inline doc/tutorials/workers.md]: A pipeline is structured in sequential stages containing one or multiple concurrent jobs.
A Job will be executed by a worker.

**Comment by proullon** [comment]
:+1:

**Pull request merged**